### PR TITLE
Feature/sources page

### LINKED ISSUE #1753: Tagging Page: Sources/Entries Filters
## Acceptance Criteria

- The filters should be expandable/collapsable
- Filter should be applied only after "apply" is clicked
- Assignee should be search multi-select input
- Add button to switch between grid and table view
- Do not persist filter

- Add only source filter for now
- Filters should be prioritized differently on different views (no need to implement for now)

### PR DESCRIPTION
**Pull request by samshara**
- fixes https://github.com/the-deep/client/issues/1753, #1754, #1752
- Depends on https://github.com/the-deep/server/pull/808

## Changes
* Add Sources Table to display sources
* Show summary of sources 
* Add Sources filter to filter sources table

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    const handleSelection = useCallback((value: boolean, name: number) => {
+        if (value) {
+            setSelectedIds(ids => ([...ids, name]));
+        } else {
+            setSelectedIds(ids => ids.filter(v => v !== name));
+        }
+    }, []);
+
+    const handleEdit = useCallback(() => {
+        console.warn('handleEdit', handleEdit);
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: Use edit modal that we have right now.
Maybe re-create the modal as well?
If too time consuming, lets create an issue for that.

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        > = {
+            id: 'sourceDetail',
+            title: 'Publisher',
+            headerCellRenderer: TableHeaderCell,
+            headerCellRendererParams: {
+                sortable: false,
+            },
+            cellRenderer: Link,
+            cellRendererParams: (_, data) => ({
+                children: data.sourceDetail.title,
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: This won't always be there.
```suggestion
                children: data.sourceDetail?.title ?? data.sourceRaw,
```

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    const columns = useMemo(() => {
+        const selectColumn: TableColumn<
+            Lead, number, CheckboxProps<number>, CheckboxProps<number>
+        > = {
+            id: 'select',
+            title: '',
+            headerCellRenderer: Checkbox,
+            headerCellRendererParams: {
+                value: selectedIds.length === leadsResponse?.results.length,
+                label: `${selectedIds.length > 0 ? `${selectedIds.length} selected` : 'Select all'}`,
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+                value: selectedIds.some(v => v === data.id),
+                onChange: handleSelection,
+            }),
+
+        };
+        const statusColumn: TableColumn<
+            Lead, number, TagProps, TableHeaderCellProps
+        > = {
+            id: 'status',
+            title: 'Status',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                actions: statusIconMap[data.status],
+                variant: statusVariantMap[data.status],
+                children: statusLabelMap[data.status],
+            }),
+        };
+        const createdAtColumn: TableColumn<
+            Lead, number, DateOutputProps, TableHeaderCellProps
+        > = {
+            id: 'createdAt',
+            title: 'Created At',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+            cellRenderer: DateOutput,
+            cellRendererParams: (_, data) => ({
+                value: data.createdAt,
+            }),
+        };
+        const publishedOnColumn: TableColumn<
+            Lead, number, DateOutputProps, TableHeaderCellProps
+        > = {
+            id: 'publishedOn',
+            title: 'Publishing Date',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+            cellRenderer: DateOutput,
+            cellRendererParams: (_, data) => ({
+                value: data.publishedOn,
+            }),
+        };
+        const publisherColumn: TableColumn<
+            Lead, number, LinkProps, TableHeaderCellProps
+        > = {
+            id: 'sourceDetail',
+            title: 'Publisher',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+            cellRendererParams: (_, data) => ({
+                children: data.sourceDetail.title,
+                to: '#', // TODO use provided url
+            }),
+        };
+        const actionsColumn: TableColumn<
+            Lead, number, ActionsProps<number>, TableHeaderCellProps
+        > = {
+            id: 'actions',
+            title: '',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx:207]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+                onEditClick: handleEdit,
+            }),
+        };
+        return ([
+            selectColumn,
+            statusColumn,
+            createdAtColumn,
+            createStringColumn<Lead, number>(
+                'title',
+                'Title',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+                item => item?.title,
+            ),
+            createStringColumn<Lead, number>(
+                'pageCount',
+                'Pages',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+                item => `${item?.pageCount} ${item?.pageCount > 1 ? 'pages' : 'page'}`,
+            ),
+            publisherColumn,
+            createStringColumn<Lead, number>(
+                'authorsDetail',
+                'Authors',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 12]
```diff
@@ -0,0 +1,279 @@
...
+                item => item?.authorsDetail.map(v => v.title).join(','),
+            ),
+            publishedOnColumn,
+            createStringColumn<Lead, number>(
+                'createdByName',
+                'Added By',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 13]
```diff
@@ -0,0 +1,279 @@
...
+                item => item?.createdByName,
+            ),
+            createStringColumn<Lead, number>(
+                'assigneeDetails',
+                'Assignee',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx [hunk 14]
```diff
@@ -0,0 +1,279 @@
...
+                item => item?.assigneeDetails.displayName,
+            ),
+            createStringColumn<Lead, number>(
+                'priorityDisplay',
+                'Priority',
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/Sources/SourcesTable/index.tsx]: use strings

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/newViews/Tagging/Sources/SourcesTable/index.tsx (hunk 1 above)
Comment (samshara) [reply src/newViews/Tagging/Sources/SourcesTable/index.tsx]: @AdityaKhatri let's create a new issue for all the actions on the sources table.

**Pull request merged**